t/lib/OpenQA/Test/Utils.pm

```diff
@@ -623,4 +623,10 @@ sub simulate_load ($load, $from) {
     return $load_avg_file;
 }
 
+sub cleanup_on_exit ($file_path) {
```

Comment:
That is a nice idea.
But avoiding to write to the git directory in the first place seems to be an easier solution, as we don't have to take care of individual files.